tests/end_to_end.rs

```diff
@@ -1832,29 +1861,56 @@ async fn test_table_time_travel() {
-        "| table_schema | table_name | column_name      | data_type                   |",
-        "+--------------+------------+------------------+-----------------------------+",
-        "| public       | test_table | some_bool_value  | Boolean                     |",
-        "| public       | test_table | some_int_value   | Int64                       |",
-        "| public       | test_table | some_other_value | Decimal128(38, 10)          |",
-        "| public       | test_table | some_time        | Timestamp(Nanosecond, None) |",
-        "| public       | test_table | some_value       | Float32                     |",
-        "+--------------+------------+------------------+-----------------------------+",
+        "+--------------+----------------+------------------+-----------------------------+",
+        "| table_schema | table_name     | column_name      | data_type                   |",
+        "+--------------+----------------+------------------+-----------------------------+",
+        "| system       | table_versions | table_schema     | Utf8                        |",
+        "| system       | table_versions | table_name       | Utf8                        |",
+        "| system       | table_versions | table_version_id | Int64                       |",
+        "| system       | table_versions | creation_time    | Timestamp(Second, None)     |",
+        "| public       | test_table     | some_bool_value  | Boolean                     |",
+        "| public       | test_table     | some_int_value   | Int64                       |",
+        "| public       | test_table     | some_other_value | Decimal128(38, 10)          |",
+        "| public       | test_table     | some_time        | Timestamp(Nanosecond, None) |",
+        "| public       | test_table     | some_value       | Float32                     |",
+        "+--------------+----------------+------------------+-----------------------------+",
+    ];
+    assert_batches_eq!(expected, &results);
+
+    //
+    // Verify that the new table versions are shown in the corresponding system table
+    //
+    let plan = context
+        .plan_query("SELECT table_schema, table_name, table_version_id FROM system.table_versions")
+        .await
+        .unwrap();
+    let results = context.collect(plan).await.unwrap();
+
+    let expected = vec![
+        "+--------------+------------+------------------+",
+        "| table_schema | table_name | table_version_id |",
+        "+--------------+------------+------------------+",
+        "| public       | test_table | 1                |",
+        "| public       | test_table | 2                |",
+        "| public       | test_table | 3                |",
+        "| public       | test_table | 4                |",
+        "| public       | test_table | 5                |",
+        "+--------------+------------+------------------+",
```

Comment:
Another test here that filters on a version ID / creation timestamp, just to make sure the rest of DataFusion projection / filtering machinery works as expected (which it should, since `TableVersionsTable` implements the table trait)?